general/communication.hpp

```diff
@@ -19,31 +19,30 @@
 #include "array.hpp"
 #include "table.hpp"
 #include "sets.hpp"
+#include "globalostream.hpp"
+#include "globalcomm.hpp"
 #include <mpi.h>
 
+
 namespace mfem
 {
 
 /** @brief A simple convenience class that calls MPI_Init() at construction and
     MPI_Finalize() at destruction. It also provides easy access to
-    MPI_COMM_WORLD's rank and size. */
+    to the global_mpi_comm rank and size. */
 class MPI_Session
 {
 protected:
    int world_rank, world_size;
-   void GetRankAndSize()
-   {
-      MPI_Comm_rank(MPI_COMM_WORLD, &world_rank);
-      MPI_Comm_size(MPI_COMM_WORLD, &world_size);
-   }
+   void GetRankAndSize();
 public:
    MPI_Session() { MPI_Init(NULL, NULL); GetRankAndSize(); }
    MPI_Session(int &argc, char **&argv)
    { MPI_Init(&argc, &argv); GetRankAndSize(); }
    ~MPI_Session() { MPI_Finalize(); }
-   /// Return MPI_COMM_WORLD's rank.
+   /// Return global_mpi_comm's rank.
    int WorldRank() const { return world_rank; }
-   /// Return MPI_COMM_WORLD's size.
+   /// Return global_mpi_comm's size.
```

Comment:
`MFEM_COMM_WORLD` instead of `global_mpi_comm`?

Reply:
Done in Veselin's update.